### ISSUE TITLE
File storage with less overhead

### ISSUE THREAD

**Issue by kripken** [kripken/emscripten#3169]
When there are very large amounts of data, the emulated filesystem can add significant overhead, as it stores another copy of the data. We should add APIs to allow storing files persistently without the filesystem, just directly using IndexedDB. The cost, however, is that these methods will be asynchronous (but perhaps using the emterpreter or asyncify, we can mitigate that).

**Comment by kripken** [comment]
I merged no-fs into incoming now. This brings:

1. An `emscripten_idb_*` API for direct IndexedDB usage, no overhead.
2. The emterpreter's async support now includes sync versions of those `_idb_` methods, as well as `emscripten_wget_data`, allowing synchronous fetching of remote data as well as synchronous caching locally.
3. I also added emterpreter support for up to 64K locals (before was 256), which allows the entire test suite to pass in both `-O3` and `-O1` modes in the emterpreter, making it more practical to use on a big project.

Overall these should get us much of the way towards a partial solution here. One big thing i want to do though is to enable better error reporting for async errors, like a sync operation when something dangerous is on the stack. Not sure how to do that yet.
